code/languages/C++/Friend_function/README.md

```diff
@@ -0,0 +1,73 @@
+## friend Function
+
+A friend function of a class is defined outside that class' scope but it has the right to access all private and protected members of the class. Even though the prototypes for friend functions appear in the class definition, friends are not member functions.
+A friend can be a function, function template, or member function, or a class or class template, in which case the entire class and all of its members are friends.
+
+To declare a function as a friend of a class, precede the function prototype in the class definition with keyword friend as follows −
+
+Declaration of friend function in C++
+``` C++
+class class_name
+{
+    ... .. ...
+    friend return_type function_name(argument/s);
+    ... .. ...
+}
+```
+Now, you can define the `friend` function as a normal function to access the data of the class. No friend keyword is used in the definition.
+
+```
+class className
```

Comment:
Add a c++ annotation. Also for all class names, first letter should be uppercase. PascalCase.